### DISCUSSION TITLE
How to use the analyse function without Internet？

### DISCUSSION THREAD

**Discussion by ninianxing** [DependencyTrack/dependency-track#1951] [Q&A]
Hi,
We have built a dependency track and NVD database in our private network, but it cannot analyse the SBOM and show vulnerabilities of component.
We built a dependency track container and download the vulnerabilities data in Internet, then move them in our private network. I have tried to configure the NVD address in our network, but the api container logs show 'MavenMetAnalyzer request failure'.
We want to analyze SBOM in dependency track without Internet.
Look forward your reply and thank you.
Attachment,
1.errlog
![errlog](https://user-images.githubusercontent.com/103549973/189802794-a18d5c46-e7aa-4991-bab9-20fc40df36d1.jpg)
2.inner vulnerability shows
![vulnerability](https://user-images.githubusercontent.com/103549973/189802818-eda801d7-6665-4e7f-8950-13ce2f35db69.jpg)
3.NVD configure
![NVDconfigure](https://user-images.githubusercontent.com/103549973/189802826-0974e4f1-5e90-4cc2-8ca1-ef270c7e192a.jpg)

**Comment by nscuro** [comment]
The meta analyzers reach out to a selection of predefined repositories in order to check for outdated components.

You can disable all public repositories in the settings, that will get rid of the errors you see in the logs.

![Screenshot 2022-09-13 at 17 54 52](https://user-images.githubusercontent.com/5693141/189949382-3735e665-9077-46d6-a66c-47965590d281.png)

The vulnerability analysis functionality should work just fine, and the errors you've been seeing are unrelated to that.

**Reply by ninianxing** [reply]
Thank you for your suggestion.
As you said, the logs not relate the analysis function. I have deleted these websites, and I meet new questions.
It will connect 'http://epss.cyentia.com' and download failed. I have disabled OSS index and vulnerDB. I want to connect our NVD in private network.
It seems downloaded NVD data successfully, but still cannot use the analysis function. I found that there is no CPE info in component. 
![微信图片_20220914111911](https://user-images.githubusercontent.com/103549973/190052098-319fb633-ed9b-4a99-a2ba-e7aef8a87c97.jpg)
The logs say 'An I/0 error has occurred while  writing a response message entity to the container output stream...broken pipe'.
![橙讯图片_1663128315555](https://user-images.githubusercontent.com/103549973/190057081-3379fe83-168e-4d18-afe9-c60bef15ffe6.jpeg)
